clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java

```diff
@@ -55,12 +58,17 @@
 
   public static synchronized TachyonFileSystem get() {
     if (sTachyonFileSystem == null) {
-      sTachyonFileSystem = new TachyonFileSystem();
+      boolean enableLineage = ClientContext.getConf().getBoolean(Constants.USER_ENABLE_LINEAGE);
+      if (enableLineage) {
+        return TachyonLineageFileSystem.get();
+      } else {
+        sTachyonFileSystem = new TachyonFileSystem();
+      }
```

Comment:
Okay, after second-thought, I feel a factory class will be cleaner than getting instance from base class.